include/hip/devicelib/sync_and_util.hh

```diff
@@ -25,6 +25,17 @@
 
 #include <hip/devicelib/macros.hh>
 
+#ifndef DEFAULT_WARP_SIZE
+// Warpsize would optimally be a device-specific, queried and made effective
+// at runtime. However, we need to fix the warpsize since SPIR-Vs need to be
+// portable across multiple devices. It should be more portable to fix it to 32
+// instead of 64, with a trade-off of potential underutilization with GPUs
+// with 64 wide warps.
+#define DEFAULT_WARP_SIZE 32
```

Comment:
Would `#error`be better here? There is a risk of having the warp size to be out of sync.

Reply:
Yes, you are right, but the problem is hipcc, we should update it to also -DDEFAULT_WARP_SIZE=X and I wanted to avoid touching that in this PR for now.